src/napari_imagej/widgets/results.py

```diff
@@ -0,0 +1,211 @@
+"""
+A QWidget designed to list SciJava SearchResults.
+
+SearchResults are grouped by the SciJava Searcher that created them.
+"""
+import atexit
+from threading import Thread
+from typing import List, Optional
+
+from jpype import JArray, JImplements, JOverride
+from qtpy.QtCore import Qt, Signal, Slot
+from qtpy.QtWidgets import QTreeWidget, QTreeWidgetItem
+
+from napari_imagej.setup_imagej import ensure_jvm_started, ij, jc
+
+
+class ResultTreeItem(QTreeWidgetItem):
+    """
+    A QTreeWidgetItem wrapping a org.scijava.search.SearchResult
+    """
+
+    def __init__(self, result: "jc.SearchResult"):
+        super().__init__()
+        self.name = ij().py.from_java(result.name())
+        self.result = result
+
+        # Set QtPy properties
+        self.setText(0, self.name)
+
+
+class SearchEventWrapper:
+    """
+    Python Class wrapping org.scijava.search.SearchEvent.
+    Needed for SearchTree.process, as signal types must be Python types.
+    """
+
+    def __init__(self, searcher: "jc.Searcher", results: List["jc.SearchResult"]):
+        self.searcher = searcher
+        self.results = [ResultTreeItem(r) for r in results]
+
+
+class SearcherTreeItem(QTreeWidgetItem):
+    """
+    A QTreeWidgetItem wrapping a org.scijava.search.Searcher
+    with a set of org.scijava.search.SearchResults
+    """
+
+    def __init__(self, searcher: "jc.Searcher"):
+        super().__init__()
+        self.title = ij().py.from_java(searcher.title())
+        self._searcher = searcher
+
+        # Set QtPy properties
+        self.setText(0, self.title)
+        self.setFlags(self.flags() & ~Qt.ItemIsSelectable)
+
+    def update(self, results: List[SearchEventWrapper]):
+        """
+        Update children with the results stored in the SearchEventWrapper
+        """
+        self.takeChildren()
+        if results and len(results):
+            self.addChildren(results)
+        self.setText(0, f"{self.title} ({len(results)})")
+        self.setExpanded(len(results) < 10)
+
+
+class ResultsTree(QTreeWidget):
```

Comment:
The naming of `ResultsTree` for the tree, versus `ResultTreeItem` for its leaf items (setting aside the `SearcherTreeItem` non-leaf items for the moment), to be a little unfortunate. It's true that the tree is for *all results* while the leaf item is for *one result*, so the inconsistency of pluralization is logical. But I wonder if we can avoid terminology that generates this inconsistency altogether.

Reply:
I think all of this has been renamed.

* `SearchResultTree` is the main tree
* `SearcherTreeItem` is the non-leaf node containing a `Searcher`
* `SearchResultTreeItem` is the leaf node class containing a `SearchResult`.